pinot-common/src/test/java/org/apache/pinot/common/utils/RegexpPatternConverterUtilsTest.java

```diff
@@ -125,4 +125,27 @@ public void testTrailingSize2() {
     String regexpLikePattern = RegexpPatternConverterUtils.likeToRegexpLike("z%");
     assertEquals(regexpLikePattern, "^z");
   }
+
+  @Test
+  public void testEscapedWildcard1() {
+    String regexpLikePattern = RegexpPatternConverterUtils.likeToRegexpLike("a\\_b_\\");
```

Comment:
Is it correct that "b_" is translated "b."? Even it is correctly, it is not obvious. Can you add some comments why it is converted as such?

Reply:
Sure, added in-line comments to explain this.